zone/mob.cpp

```diff
@@ -549,10 +549,17 @@ void Mob::SetInvisible(uint8 state)
 	invisible = state;
 	SendAppearancePacket(AT_Invis, invisible);
 	// Invis and hide breaks charms
-
 	auto formerpet = GetPet();
-	if (formerpet && formerpet->GetPetType() == petCharmed && (invisible || hidden || improved_hidden))
-		formerpet->BuffFadeByEffect(SE_Charm);
+	if (formerpet && formerpet->GetPetType() == petCharmed && (invisible || hidden || improved_hidden || invisible_animals || invisible_undead)) {
+		if (RuleB(Pets, LivelikeBreakCharmOnInvis)) {	//Pets:LivelikeBreakCharmOnInvis = true
```

Comment:
Don't really need this comment here.

Reply:
Agree with @hgtw the `IsInvisible` function should simplify and cover all of the OR's in this line